modules/javafx.base/src/main/java/javafx/beans/value/MappedBinding.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2021 Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package javafx.beans.value;
+
+import java.util.Objects;
+import java.util.function.Function;
+
+import com.sun.javafx.binding.Subscription;
+
+class MappedBinding<S, T> extends LazyObjectBinding<T> {
+    private final ObservableValue<S> source;
+    private final Function<? super S, ? extends T> mapper;
+
+    public MappedBinding(ObservableValue<S> source, Function<? super S, ? extends T> mapper) {
+        this.source = Objects.requireNonNull(source, "source cannot be null");
+        this.mapper = Objects.requireNonNull(mapper, "mapper cannot be null");
+    }
+
+    @Override
+    protected Subscription observeInputs() {
+        return Subscription.subscribeInvalidations(source, this::invalidate); // start observing source
+    }
+
+    @Override
+    protected T computeValue() {
```

Comment:
Maybe move this method above `observeInputs` like in `FlatMapBinding`. Same for `OrElseBinding`.